plugins/bap/utils/ida.py

```diff
@@ -99,6 +99,68 @@ def func_name_propagate_thunk(ea):
                     GetFunctionAttr(f, FUNCATTR_END)))
 
 
+def dump_c_header(output_filename):
+    """Dump type information as a C header."""
+    def local_type_info():
+        class my_sink(idaapi.text_sink_t):
+            def __init__(self):
+                try:
+                    idaapi.text_sink_t.__init__(self)
+                except AttributeError:
+                    pass  # Older IDA versions keep the text_sink_t abstract
+                self.text = []
+
+            def _print(self, thing):
+                self.text.append(thing)
+                return 0
+
+        sink = my_sink()
+
+        idaapi.print_decls(sink, idaapi.cvar.idati, [],
+                           idaapi.PDF_INCL_DEPS | idaapi.PDF_DEF_FWD)
+        return sink.text
+
+    def function_sigs():
+        import idautils
+        f_types = []
+        for ea in idautils.Functions():
+            ft = idaapi.print_type(ea, True)
+            if ft is not None:
+                f_types.append(ft + ';')
+        return list(set(f_types))  # Set, since sometimes, IDA gives repeats
+
+    def replacer(regex, replacement):
+        import re
+        r = re.compile(regex)
+        return lambda s : r.sub(replacement, s)
+
+    pp_decls = replacer(r'(struct|enum|union) ([^{} ]*);',
+                         r'\1 \2; typedef \1 \2 \2;')
+    pp_unsigned = replacer(r'unsigned __int(8|16|32|64)',
+                           r'uint\1_t')
+    pp_signed = replacer(r'(signed )?__int(8|16|32|64)',
+                         r'int\2_t')
+    pp_annotations = replacer(r'__cdecl|__noreturn', '')
```

Comment:
can you subtitute them with `__attribute__((cdecl))` and `__attribute __((noreturn))` correspondingly. Also, it might be the case, that the placement matters, although I think that it should be OK.